fabric-gametest-api-v1/src/main/java/net/fabricmc/fabric/impl/gametest/FabricGameTestHelper.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.gametest;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.function.Consumer;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import net.minecraft.resource.ResourcePackManager;
+import net.minecraft.resource.ServerResourceManager;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.test.GameTestBatch;
+import net.minecraft.test.TestContext;
+import net.minecraft.test.TestFunction;
+import net.minecraft.test.TestFunctions;
+import net.minecraft.test.TestServer;
+import net.minecraft.test.TestUtil;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.registry.DynamicRegistryManager;
+import net.minecraft.world.level.storage.LevelStorage;
+
+import net.fabricmc.fabric.api.gametest.v1.FabricGameTest;
+import net.fabricmc.loader.FabricLoader;
+
+public final class FabricGameTestHelper {
+	public static final boolean ENABLED = System.getProperty("fabric-api.gametest") != null;
+
+	private static final Logger LOGGER = LogManager.getLogger();
+
+	private FabricGameTestHelper() {
+	}
+
+	public static void runHeadlessServer(LevelStorage.Session session, ResourcePackManager resourcePackManager, ServerResourceManager serverResourceManager, DynamicRegistryManager.Impl registryManager) {
+		LOGGER.info("Starting test server");
+		MinecraftServer server = TestServer.startServer(thread -> {
+			TestServer testServer = new TestServer(thread, session, resourcePackManager, serverResourceManager, getBatches(), BlockPos.ORIGIN, registryManager);
+			FabricLoader.INSTANCE.setGameInstance(testServer);
```

Comment:
Humm, just noticed this is using the internal loader api to do this, might need checking with 0.12 first.

Reply:
This is going to break, going to back this out of last call for now untill a solution is found.